meson.build

```diff
@@ -565,13 +565,22 @@ endif
 # Let targets add more support libraries 
 additional_libs_list = meson.get_cross_property('additional_libs', [])
 
-if cc.get_id() == 'ccomp'
+compiler_id = cc.get_id()
+if compiler_id == 'ccomp'
   # When passing the specs file to CompCert, the libcompcert needs to be included there as well
   additional_libs_list += '-lcompcert'
 endif
 
 additional_libs = ' '.join(additional_libs_list)
 
+if compiler_id == 'gcc' and target_machine.cpu_family() == 'msp430'
```

Comment:
Usually it's possible to amend the compiler configuration to let it do things like this and then adjust to make picolibc work. It sounds like that's just not possible here? I guess I should go look at what gcc is doing and see if there's a way to avoid this kind of low-level kludgery. Of course, if this is what it takes, then we'll use it.

Reply:
I'm not at all married to this "solution", if you have something else that makes `-mhwmult` select the multiplication library to link, I'm happy to use that.